File: mocha-game/src/main/java/mocha/game/world/item/Item.java
package mocha.game.world.item;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import mocha.game.world.item.itemprototype.ItemPrototype;
import mocha.shared.Identified;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Item implements Identified<Integer> {

  private Integer id;
  private ItemPrototype itemPrototype;
  private Integer data0;
  private Integer data1;
  private Integer data2;

  public String getSpriteId() {
    return itemPrototype.getSpriteId();
  }

}

File: mocha-game/src/main/java/mocha/game/world/item/Item.kt
package mocha.game.world.item

import mocha.shared.Identified

open class Item : Identified<Int> {

  private var id: Int? = null
  open var itemPrototype: ItemPrototype? = null
  open var data0: Int? = null
  open var data1: Int? = null
  open var data2: Int? = null

  val spriteId: String?
    get() = itemPrototype!!.spriteId

  constructor()

  constructor(id: Int?, itemPrototype: ItemPrototype, data0: Int?, data1: Int?, data2: Int?) {
    this.id = id
    this.itemPrototype = itemPrototype
    this.data0 = data0
    this.data1 = data1
    this.data2 = data2
  }

  override fun getId(): Int? {
    return id
  }

  override fun setId(id: Int?) {
    this.id = id
  }
}
